### PR TITLE
[plugin][MLA] optimize MLA metadata build and remove D2D copy

### PR DESCRIPTION
**Pull request by zejunchen-zejun**
This PR improves MLA decode performance in ATOM's vLLM plugin mode by reducing per-step metadata work, using direct KV index generation, and removing a redundant positions device-to-device copy

For performance:
Model | Concurrency | Baseline (total token/s) | this PR (total token/s) | Gain
-- | -- | -- | -- | --
Kimi-K2 TP8 | 4 | 666.38 | 678.63 | +1.84%
Kimi-K2 TP8 | 32 | 3916.37 | 3967.31 | +1.30%
DeepSeek-R1 FP8 TP8 | 4 | 630.41 | 641.05 | +1.69%
DeepSeek-R1 FP8 TP8 | 32 | 3270.91 | 3310.07 | +1.20%

For accuracy:
Model | Baseline | this PR | Delta
-- | -- | -- | --
Kimi-K2 TP8 | 0.9325| 0.9401| +0.0076
DeepSeek-R1 FP8 TP8 | 0.9492| 0.9409| -0.0083

### CODE REVIEW

File: atom/utils/envs.py [hunk 1]
```diff
@@ -75,6 +75,10 @@
         "ATOM_DISABLE_VLLM_PLUGIN_ATTENTION", "0"
     ).lower()
     == "1",
+    "ATOM_USE_PERSISTENT_MLA_DECODE_METADATA": lambda: os.getenv(
+        "ATOM_USE_PERSISTENT_MLA_DECODE_METADATA", "1"
+    ).lower()
+    == "1",
```
[the remark this answers is not in the document]
Comment (wuhuikx) [reply atom/utils/envs.py]: yes, please add the description of the env var in the **docs/environment_variables.md**

File: atom/plugin/attention_mla.py [hunk 2]
```diff
@@ -545,76 +546,97 @@ def _forward_prefill_plugin_mode(
...
     ) -> tuple[torch.Tensor, torch.Tensor | None]:
-        assert kv_c_and_k_pe_cache.numel() > 0
-        assert attn_metadata.plugin_metadata.decode is not None
-        assert attn_metadata.plugin_metadata.decode.max_qo_len is not None
-
-        # if type(q) is tuple:
-        #     q = torch.cat(q, dim=-1)
-
-        assert isinstance(q, torch.Tensor)
-        if self.head_repeat_factor > 1:
```
Comment (gbyu-amd) [inline atom/plugin/attention_mla.py]: The head repeat logic is needed for Kimi tp8. Did you remove it or refactor it somewhere else?
